deploy/eck-elastic-agent/values.yaml

```diff
@@ -0,0 +1,165 @@
+---
+# Default values for eck-elastic-agent.
+# This is a YAML-formatted file.
+
+# Overridable names of the Elastic Agent resource.
+# By default, this is the Release name set for the chart,
+# followed by 'eck-elastic-agent'.
+#
+# nameOverride will override the name of the Chart with the name set here,
+# so nameOverride: quickstart, would convert to '{{ Release.name }}-quickstart'
+#
+# nameOverride: "quickstart"
+#
+# fullnameOverride will override both the release name, and the chart name,
+# and will name the Fleet Agent resource exactly as specified.
+#
+# fullnameOverride: "quickstart"
+
+# Version of Elastic Agent.
+#
+version: 8.2.3
+
+# Labels that will be applied to Elastic Agent.
+#
+labels: {}
+
+# Annotations that will be applied to Elastic Agent.
+#
+annotations: {}
+
+spec:
+  # Referenced resources are below and depending on the setup, at least one is required for a functional Agent.
+  # ref: https://www.elastic.co/guide/en/cloud-on-k8s/current/k8s-elastic-agent-fleet-configuration.html#k8s-elastic-agent-fleet-configuration-setting-referenced-resources
+  #
+  # Reference to ECK-managed Kibana instance.
+  #
+  # kibanaRef:
+  #   name: quickstart
+    # Optional namespace reference to Kibana instance.
+    # If not specified, then the namespace of the Agent instance
+    # will be assumed.
+    #
+    # namespace: default
+
+  # Reference to ECK-managed Elasticsearch instance.
+  #
+  elasticsearchRefs:
+  - name: elasticsearch
+    # Optional namespace reference to Elasticsearch instance.
+    # If not specified, then the namespace of the Agent instance
+    # will be assumed.
+    #
+    # namespace: default
+  
+  # Reference to ECK-managed Fleet Server instance.
+  #
+  # fleetServerRef:
+  #   name: fleet-server
+    # Optional namespace reference to Fleet Server instance.
+    # If not specified, then the namespace of the Agent instance
+    # will be assumed.
+    #
+    # namespace: default
+
+  # Daemonset, or Deployment specification for Agent.
+  # At least one is required of [daemonSet, deployment].
+  # ref: https://www.elastic.co/guide/en/cloud-on-k8s/current/k8s-elastic-agent-fleet-configuration.html#k8s-elastic-agent-fleet-configuration-chose-the-deployment-model
+  #
+  daemonSet:
+    podTemplate:
+      spec:
+        containers:
+        - name: agent
+          securityContext:
+            runAsUser: 0
+
+  # Configuration of Agent, specifically used in Agent standalone mode.
+  # ref: https://www.elastic.co/guide/en/cloud-on-k8s/current/k8s-elastic-agent.html
+  #
+  config: {}
```

Comment:
Should we let the user set `config` and `elasticsearchRefs ` explicitly? In fleet mode, using the provided example `deploy/eck-elastic-agent/examples/fleet-agents.yaml`,  it fails with:

```
Error: INSTALLATION FAILED: admission webhook "elastic-agent-validation-v1alpha1.k8s.elastic.co" denied the request: Agent.agent.k8s.elastic.co "my-agents-eck-elastic-agent" is invalid:
[spec.config: Invalid value: v1.Config{Data:map[string]interface {}{}}: remove config, it can't be set in fleet mode, spec.fleetServerEnabled: Invalid value: false: remove Elasticsearch reference, it can't be enabled in fleet mode when Fleet Server is not enabled as well]
```

Reply:
I've disabled them explicitly in the examples now.  We have to allow these to be adjustable, as the agent chart supports both fleet, and standalone, so config is required in some cases.  Are you thinking that config should just be set to `null` in the default values instead of `{}`, and the ES ref left empty?